src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -248,8 +248,10 @@ public final void close() throws SQLException
             }
          }
          finally {
-            delegate = ClosedConnection.CLOSED_CONNECTION;
-            poolEntry.recycle(lastAccess);
+            if (delegate != ClosedConnection.CLOSED_CONNECTION) {
+               delegate = ClosedConnection.CLOSED_CONNECTION;
+               poolEntry.recycle(lastAccess);
+            }
```

Comment:
this avoids 'possibility' of adding to the bag and worse 'giving user' a ClosedConnection.CLOSED_CONNECTION.